tests/integration_tests/releases.py

```diff
@@ -0,0 +1,93 @@
+import functools
+import logging
+from typing import Optional
+
+from packaging import version
+
+from cloudinit import subp
+from cloudinit.subp import ProcessExecutionError
+from tests.integration_tests import integration_settings
+
+log = logging.getLogger("integration_testing")
+
+
+def get_all_ubuntu_series() -> list:
+    """Use distro-info-data's ubuntu.csv to get a list of Ubuntu series"""
+    out = ""
+    try:
+        out, _err = subp.subp(["ubuntu-distro-info", "-a"])
+    except ProcessExecutionError:
+        log.info(
+            "ubuntu-distro-info (from the distro-info package) must be"
+            " installed to guess Ubuntu os/release"
+        )
+    return out.splitlines()
+
+
+def ubuntu_version_from_series(series) -> str:
+    try:
+        out, _err = subp.subp(
+            ["ubuntu-distro-info", "--release", "--series", series]
+        )
+    except subp.ProcessExecutionError as e:
+        raise ValueError(
+            f"'{series}' is not a recognized Ubuntu release"
+        ) from e
+    return out.rstrip(" LTS")
+
+
+@functools.total_ordering
+class Release:
+    def __init__(
+        self,
+        os: str,
+        release: str,
+        version: str,
+        image_id: Optional[str] = None,
+    ):
+        self.os = os
+        self.series = release
+        self.version = version
+        self.image_id = image_id
+
+    def __repr__(self):
+        return f"Release({self.os}, {self.version})"
+
+    def __lt__(self, other: "Release"):
+        if self.os != other.os:
+            return False
```

Comment:
From [the docs](https://docs.python.org/3/reference/datamodel.html#object.__lt__):

> A rich comparison method may return the singleton NotImplemented if it does not implement the operation for a given pair of arguments. By convention, False and True are returned for a successful comparison.

I think we should probably shouldn't return `False` when comparing `ubuntu` to `debian`.

Reply:
Good point, I was thinking more equality. An exception seems more appropriate here.

Reply:
Thanks. I think it actually makes more sense to raise an exception here rather than return NotImplemented:
```
Traceback (most recent call last):
  File "/home/james/unrelated_python.py", line 3, in <module>
    print(releases.LUNAR > releases.SOMETHING)
TypeError: '>' not supported between instances of 'Release' and 'Release'
```

vs

```
Traceback (most recent call last):
  File "/home/james/unrelated_python.py", line 3, in <module>
    print(releases.LUNAR > releases.SOMETHING)
  File "/home/james/.pyenv/versions/3.7.13/lib/python3.7/functools.py", line 93, in _gt_from_lt
    op_result = self.__lt__(other)
  File "/home/james/c/cloud-init/tests/integration_tests/releases.py", line 58, in __lt__
    raise ValueError(f"{self.os} cannot be compared to {other.os}!")
ValueError: ubuntu cannot be compared to debian!
```

Reply:
Agreed, I think that looks better.